app/core/pokemon-entity.ts

```diff
@@ -435,11 +435,14 @@ export class PokemonEntity extends Schema implements IPokemonEntity {
     value: number,
     caster: IPokemonEntity,
     apBoost: number,
-    crit: boolean
+    crit: boolean,
+    permanent = false
   ) {
     value =
       value * (1 + (apBoost * caster.ap) / 100) * (crit ? caster.critPower : 1)
-    this.hp = min(1)(this.hp + value)
+    const target = permanent ? this.refToBoardPokemon : this
+    target.hp = min(1)(target.hp + value)
+    if(permanent) return
```

Comment:
you forget to check for ghost opponents which was the initial point of this PR
Also there's no scenario where we want to add a stat on the board pokemon but not the entity, so there's no point for an early return here